### ISSUE TITLE
[req]: 

### ISSUE THREAD

**Issue by batuhankrmn** [Jackett/Jackett#13640]
### Is there already a request for your feature?

- [X] I have checked older issues, open and closed

### Is your feature request related to a problem? Please describe.

I have multiple saved filters. Let's say I searched for some movie. I want to filter the results for x264 and x265 respectively. If I select x264 filter, I can't select the x265 filter without deleting the x264 filter because dropdown menu shows only the selected filter. Also there is no "clear" button for filter field. So we need to select filter text, delete it with backspace key and select other filter. I believe this should be improved in terms of the UX.

![image](https://user-images.githubusercontent.com/57859656/195448739-eb097128-7efc-45d9-98df-891e2ea1c035.png)
![image](https://user-images.githubusercontent.com/57859656/195448844-4f325385-f02c-4abc-b3a5-cab23330fc32.png)

### Describe the solution you'd like

- [ ] When the user selects a saved filter from the dropdown menu, we should also show other saved filters in the dropdown menu.
- [ ] To make it easier to switch between filters and see the results unfiltered , we should add a "clear" button to clear the search filter field.

### Describe alternatives you've considered

I don't think it's logical and useful to have a dropdown menu right below the search filtering box. Instead, a new menu can be added to the right side of the Category dropdown menu, which will ensure design integrity and improve the user experience. 

![image](https://user-images.githubusercontent.com/57859656/195451116-87d99e78-03ea-4b0e-bfbd-e2895c851bf5.png)

**Comment by garfield69** [comment]
>I don't think it's logical and useful to have a dropdown menu right below the search filtering box. Instead, a new menu can be added to the right side of the Category dropdown menu,

I disagree with this.
where it is now is helps disassociate it with the query row above, since that row deals with preparing the query to the indexer(s)
whereas where it is now deals with filtering the results once the have been fetched from the indexers.

**Comment by garfield69** [comment]
we could rename the `Search:` text to `Search Results:` to make it more clearer that this deals with filtering after results have been fetched.

**Comment by batuhankrmn** [comment]
> > I don't think it's logical and useful to have a dropdown menu right below the search filtering box. Instead, a new menu can be added to the right side of the Category dropdown menu,
> 
> I disagree with this. where it is now is helps disassociate it with the query row above, since that row deals with preparing the query to the indexer(s) whereas where it is now deals with filtering the results once they have been fetched from the indexers.

I see your point, what I mean is that the dropdown menu under the search filtering box gives me the feeling that it is difficult to improve. and as it stands now, it doesn't fit the Jackett design. I don't know the logic behind it, but visually this dropdown menu is the same as the one in Chrome that shows the usernames previously entered.

**Comment by garfield69** [comment]
yeah, the filter list presentation does look a little odd, I originally expected it to look like the box itself was expanding to show the list, rather than a separate popup box below and apart of the search box itself.
room for improvement there but will likely require someone with GUI and html/css skills.

**Comment by ilike2burnthing** [comment]
For the initial req, filtering the filters by what is currently typed (the current behaviour) seems like the best approach to me. If someone has 100 filters, 10 include `x264`, they can then reduce a long list of 100 down to 10 very quickly by just typing `x264`.

As for the addition of a `Clear` button, I don't really see the need, highlight and delete, or triple-click and delete, or Ctrl+A and delete all work fine and don't clutter the UI.

I have no issue changing `Search:` to `Search Results:` or `Filter:` or `Filter Results:`, or whatever seems clearest.

The drop down box of saved filters appears to be browser dependent. Above is Chrome, whereas Edge and Opera use:

![edge](https://user-images.githubusercontent.com/59480337/195457278-9390ae51-49b0-4eaf-9077-2161cb1fa694.png)

and Firefox uses:

![firefox](https://user-images.githubusercontent.com/59480337/195457308-aad6f627-53d1-4cc9-a2c1-0b757b0a1eaf.png)

Not sure if we can alter that.

**Comment by batuhankrmn** [comment]
> If someone has 100 filters, 10 include `x264`, they can then reduce a long list of 100 down to 10 very quickly by just typing `x264`.

This would work great but let's think like that: someone search for some movies and want to check for resolution options. He/she has saved filters such as `720p` `1080p` `2160p` . With the current working logic, switching between these 3 resolution filters is quite tricky. `Select 720p --> Triple Click --> Delete --> Click Search Box --> 1080p --> Triple Click --> Delete --> Click Search Box --> 2160p  --> Triple Click --> Delete` 
In the working logic I propose, you can switch between filters with just 3 clicks without even taking your hand off the mouse and then remove all filtering with the "Clear" button.

**Comment by garfield69** [comment]
>The drop down box of saved filters appears to be browser dependent.

that's interesting. on my win10 chrome browser 106 i get
![image](https://user-images.githubusercontent.com/6933240/195460093-2d13ad9b-84b6-4154-8ef4-d466374bcb20.png)

where as I was expecting what you see, the fill length of the box pulled down.

**Comment by ilike2burnthing** [comment]
@garfield69 just Chrome trying to be 'fancy' basically, don't know if we can do much about it.

@batuhankrmn currently users can filter filters but have to use a minor amount of effort to change filters. With your proposed way, users could no longer filter filters but can change filters slightly easier. I'm going to side with greater functionality at the cost an extra couple of clicks.

**Comment by batuhankrmn** [comment]
@ilike2burnthing Although it is not a very common scenario to have 100 filters and filter them in itself, I agree  it's the function. In this case, since we can't have both features at the same time, wouldn't adding a "Clear" button at least improve the situation? 

With the "Clear" button, the selected filter can be quickly cleared and a new one can be selected. (After clicking the Clear button, the dropdown menu will be opened automatically)

**Comment by ilike2burnthing** [comment]
I don't see the point of cluttering the UI with another button to save a couple of clicks.

**Comment by batuhankrmn** [comment]
Of course, I presented my proposal in terms of UX, the rest is up to you ✌🏼

**Comment by ilike2burnthing** [comment]
Then this issue just remains open if @garfield69 feels that the dropdown box can be made more uniform across different browsers, and/or to change `Search:` to something more fitting. If the former, then the title would need edited.

**Comment by mikeoscar2006** [comment]
I think the UI has to be changed for that part atleast. As I mentioned, it was more of a PoC from my side about what can happen. A UI/UX Designer can maybe convert it to bring uniformity and make it fit with Jackett's current UI/UX.

**Comment by ngosang** [comment]
> Then this issue just remains open if @garfield69 feels that the dropdown box can be made more uniform across different browsers, and/or to change Search: to something more fitting. If the former, then the title would need edited.

I agree with both things. Keep current functionality and make those usability improvements.
